<a href="https://colab.research.google.com/github/SnoringBoy/MyfirstApp/blob/master/KDD_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## KDD **Project: Context-Aware Multi-Modal Transportation Recommendation**
[Project Link](https://dianshi.baidu.com/competition/29/rule)

# New Section

## Download data , and Merge data

In [0]:
# add libraries 
import csv
import pandas as pd
import os

In [3]:
print('memory that we can use')
!cat /proc/meminfo | grep 'MemAvailable'
print('if it had turbo boost it would''ve shown Min and Max MHz also but it is only showing current frequency this means it always operates at 2.3GHz')
!lscpu | grep "MHz"
print('hard disk that we can use')
!df -h / | awk '{print  $4}'

memory that we can use
MemAvailable:   12671212 kB
if it had turbo boost it wouldve shown Min and Max MHz also but it is only showing current frequency this means it always operates at 2.3GHz
CPU MHz:             2300.000
hard disk that we can use
Avail
26G


In [4]:
#get data from dropbox
!wget https://dl.dropboxusercontent.com/s/6hwa4gpqn5jcrev/data_set_phase1.zip

--2019-04-15 21:59:02--  https://dl.dropboxusercontent.com/s/6hwa4gpqn5jcrev/data_set_phase1.zip
Resolving dl.dropboxusercontent.com (dl.dropboxusercontent.com)... 162.125.1.6, 2620:100:6016:6::a27d:106
Connecting to dl.dropboxusercontent.com (dl.dropboxusercontent.com)|162.125.1.6|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 49277572 (47M) [application/zip]
Saving to: ‘data_set_phase1.zip’

data_set_phase1.zip 100%[===================>]  46.99M  54.9MB/s    in 0.9s    

2019-04-15 21:59:04 (54.9 MB/s) - ‘data_set_phase1.zip’ saved [49277572/49277572]



In [5]:
# unzip dataset zip file
!unzip data_set_phase1.zip 

Archive:  data_set_phase1.zip
   creating: data_set_phase1/
  inflating: data_set_phase1/train_queries.csv  
  inflating: data_set_phase1/test_plans.csv  
  inflating: data_set_phase1/test_queries.csv  
  inflating: data_set_phase1/train_clicks.csv  
  inflating: data_set_phase1/profiles.csv  
  inflating: data_set_phase1/train_plans.csv  


In [6]:
cd data_set_phase1/

/content/data_set_phase1


In [7]:
ls

profiles.csv     test_queries.csv*  train_plans.csv*
test_plans.csv*  train_clicks.csv*  train_queries.csv*


In [8]:
# information of profiles.csv  # 1
profiles = pd.read_csv('profiles.csv')
List= list(profiles)
print(profiles.shape)
print(*List ,sep = "\n")


(63090, 67)
pid
p0
p1
p2
p3
p4
p5
p6
p7
p8
p9
p10
p11
p12
p13
p14
p15
p16
p17
p18
p19
p20
p21
p22
p23
p24
p25
p26
p27
p28
p29
p30
p31
p32
p33
p34
p35
p36
p37
p38
p39
p40
p41
p42
p43
p44
p45
p46
p47
p48
p49
p50
p51
p52
p53
p54
p55
p56
p57
p58
p59
p60
p61
p62
p63
p64
p65


In [9]:
profiles.head()

pid   p0   p1   p2   p3   p4   p5   p6   p7   p8 ...   p56  p57  p58  \
0  196356  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  1.0 ...   0.0  0.0  0.0   
1  204083  1.0  0.0  0.0  0.0  0.0  0.0  0.0  1.0  0.0 ...   0.0  0.0  0.0   
2  170667  0.0  0.0  1.0  0.0  0.0  0.0  0.0  1.0  0.0 ...   0.0  0.0  0.0   
3  115511  0.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0  1.0 ...   0.0  0.0  0.0   
4  129719  0.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  1.0 ...   0.0  0.0  0.0   

   p59  p60  p61  p62  p63  p64  p65  
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  
1  1.0  0.0  0.0  0.0  1.0  0.0  0.0  
2  0.0  0.0  0.0  1.0  0.0  0.0  0.0  
3  0.0  0.0  0.0  0.0  1.0  0.0  0.0  
4  0.0  1.0  0.0  0.0  0.0  0.0  0.0  

[5 rows x 67 columns]

In [10]:
for k, v in profiles.nunique().to_dict().items():
    print('{}={}'.format(k,v))

pid=63090
p0=2
p1=2
p2=2
p3=2
p4=2
p5=2
p6=2
p7=2
p8=2
p9=2
p10=2
p11=2
p12=2
p13=2
p14=2
p15=2
p16=2
p17=2
p18=2
p19=2
p20=2
p21=2
p22=2
p23=2
p24=2
p25=2
p26=2
p27=2
p28=2
p29=2
p30=2
p31=2
p32=2
p33=2
p34=2
p35=2
p36=2
p37=2
p38=2
p39=2
p40=2
p41=2
p42=2
p43=2
p44=2
p45=2
p46=2
p47=2
p48=2
p49=2
p50=2
p51=2
p52=2
p53=2
p54=2
p55=2
p56=2
p57=2
p58=2
p59=2
p60=2
p61=2
p62=2
p63=2
p64=2
p65=2


In [12]:
# information of train_plans.csv #2
train_plans = pd.read_csv('train_plans.csv')
print(train_plans.shape , list(train_plans))
for k, v in train_plans.nunique().to_dict().items():
    print('{}={}'.format(k,v))
train_plans.head() 

(491054, 3) ['sid', 'plan_time', 'plans']
sid=491054
plan_time=455427
plans=489563


sid            plan_time  \
0  1709112  2018-11-04 11:45:04   
1  3327773  2018-10-16 19:09:29   
2  3285959  2018-11-25 15:05:22   
3  1616170  2018-10-03 14:42:40   
4   351369  2018-10-05 19:52:37   

                                               plans  
0  [{"distance": 32303, "price": 600, "eta": 5087...  
1  [{"distance": 3427, "price": 300, "eta": 627, ...  
2  [{"distance": 36234, "price": 1100, "eta": 931...  
3  [{"distance": 11450, "price": 400, "eta": 2801...  
4  [{"distance": 18495, "price": 600, "eta": 4271...

In [13]:
# information of train_clicks.csv #3

train_clicks = pd.read_csv('train_clicks.csv')
print(train_clicks.shape , list(train_clicks))
for k, v in train_clicks.nunique().to_dict().items():
    print('{}={}'.format(k,v))
train_clicks.head()

(453336, 3) ['sid', 'click_time', 'click_mode']
sid=453336
click_time=422100
click_mode=11


sid           click_time  click_mode
0  2848914  2018-11-17 18:42:17           1
1  2629085  2018-10-12 16:28:13           3
2   602598  2018-11-11 16:38:42           2
3  2022975  2018-10-14 12:28:11           9
4   988425  2018-11-12 16:47:30           7

In [15]:
# information of train_queries.csv #4

train_queries = pd.read_csv('train_queries.csv')
print(train_queries.shape, list(train_queries))
for k, v in train_queries.nunique().to_dict().items():
    print('{}={}'.format(k,v))

train_queries.head()

(500000, 5) ['sid', 'pid', 'req_time', 'o', 'd']
sid=500000
pid=46191
req_time=464211
o=5460
d=4923


sid       pid             req_time             o             d
0  3000821       NaN  2018-11-02 17:54:30  116.29,39.97  116.32,39.96
1  3085857  210736.0  2018-11-16 10:53:10  116.39,39.84  116.33,39.79
2  2944522       NaN  2018-10-06 10:33:58  116.31,39.93  116.27,40.00
3   559931  202427.0  2018-11-23 14:54:11  116.27,39.88  116.39,39.90
4  2819352  172251.0  2018-10-30 11:48:41  116.34,39.96  116.37,39.86

In [16]:
# information of test_plans.csv #5

test_plans = pd.read_csv('test_plans.csv')
print(test_plans.shape , list(test_plans))
for k, v in test_plans.nunique().to_dict().items():
    print('{}={}'.format(k,v))
test_plans.head()

(92571, 3) ['sid', 'plan_time', 'plans']
sid=92571
plan_time=82432
plans=92163


sid            plan_time  \
0  1112456  2018-12-05 17:39:47   
1  1413458  2018-12-06 17:49:17   
2  1243160  2018-12-02 14:50:25   
3  2040494  2018-12-07 09:23:45   
4  1448779  2018-12-06 18:24:46   

                                               plans  
0  [{"distance": 465, "price": "", "eta": 418, "t...  
1  [{"distance": 28846, "price": 600, "eta": 4779...  
2  [{"distance": 24646, "price": 600, "eta": 3513...  
3  [{"distance": 32235, "price": 600, "eta": 4072...  
4  [{"distance": 13502, "price": 400, "eta": 3195...

In [17]:
# information of test_queries.csv #6
test_queries = pd.read_csv('test_queries.csv')
print(test_queries.shape , list(test_queries))
for k, v in test_queries.nunique().to_dict().items():
    print('{}={}'.format(k,v))
test_queries.head()

(94358, 5) ['sid', 'pid', 'req_time', 'o', 'd']
sid=94358
pid=13299
req_time=84156
o=3257
d=2884


sid       pid             req_time             o             d
0  1126541  178395.0  2018-12-05 14:51:43  116.41,39.92  116.46,39.95
1   453685  179178.0  2018-12-01 12:50:33  116.39,39.90  116.41,39.91
2  2066073  184748.0  2018-12-07 10:17:35  116.37,39.71  116.31,39.89
3   452746  162795.0  2018-12-01 07:52:44  116.33,39.97  116.31,39.98
4  1431672  116384.0  2018-12-06 10:36:30  116.51,39.68  116.51,39.68

Below are some important information related to dataset

**profiles.csv # 1 	--> 63090, 67**

**train_plans.csv #2 	--> 491054, 3**

**train_clicks.csv #3 -->453336, 3**

**train_queries.csv #4-->500000, 5**

**test_plans.csv #5  	-->92571, 3**

**test_queries.csv #6 -->94358, 5**

In [18]:
print('memory that we can use')
!cat /proc/meminfo | grep 'MemAvailable'
print('if it had turbo boost it would''ve shown Min and Max MHz also but it is only showing current frequency this means it always operates at 2.3GHz')
!lscpu | grep "MHz"
print('hard disk that we can use')
!df -h / | awk '{print  $4}'

memory that we can use
MemAvailable:   12043876 kB
if it had turbo boost it wouldve shown Min and Max MHz also but it is only showing current frequency this means it always operates at 2.3GHz
CPU MHz:             2300.000
hard disk that we can use
Avail
26G


In [0]:
2+3

5